### PR TITLE
Remove unused Makefile target

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**: This target is unused, removing it. Credit to @jackfrancis for finding this.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -354,20 +354,6 @@ create-workload-cluster: $(ENVSUBST) $(KUBECTL) ## Create a workload cluster.
 
 	@echo 'run "$(KUBECTL) --kubeconfig=./kubeconfig ..." to work with the new target cluster'
 
-.PHONY: create-aks-cluster
-create-aks-cluster: $(KUSTOMIZE) $(ENVSUBST) $(KUBECTL) ## Create a aks cluster.
-	# Create managed Cluster.
-	$(ENVSUBST) < $(TEMPLATES_DIR)/$(MANAGED_CLUSTER_TEMPLATE) | $(KUBECTL) apply -f -
```
Comment (jackfrancis) [inline Makefile:360]: can we get rid of the `MANAGED_CLUSTER_TEMPLATE` var now?

Comment (CecileRobertMichon) [reply Makefile:360]: done

**Review by jackfrancis — APPROVED** [review]
/lgtm
/approve

**Comment by jackfrancis** [comment]
lol ci-entrypoint.sh flake

**Pull request merged**